core_backend/app/llm_call/parse_input.py

```diff
@@ -116,6 +116,48 @@ async def wrapper(
     return wrapper
 
 
+def _process_identified_language_response(
+    identified_language: IdentifiedLanguage | None,
+    response: UserQueryResponse,
+    task: str = "LANGUAGE IDENTIFICATION",
+) -> UserQueryResponse | UserQueryResponseError:
+    """Process the identified language and return the response."""
+    supported_languages = IdentifiedLanguage.get_supported_languages()
+    supported_language_string = ", ".join(supported_languages)
+
+    match identified_language:
+        case IdentifiedLanguage.UNINTELLIGIBLE:
+            error_response = UserQueryResponseError(
+                error_message=STANDARD_FAILURE_MESSAGE
+                + " Unintelligible question. The following languages are supported: "
+                f"{supported_language_string}. ",
+                query_id=response.query_id,
+                error_type=ErrorType.UNINTELLIGIBLE_QUESTION,
+            )
+            logger.info(
+                f"{task} FAILED due to {identified_language.value} "
```

Comment:
For clarity, the if-condition and error message are hardcoded anyway:
```suggestion
                f"{task} FAILED due to UNINTELLIGIBLE "
```